Sources/Datadog/Core/Persistence/FileWriter.swift

```diff
@@ -20,7 +20,19 @@ internal final class FileWriter {
         self.orchestrator = orchestrator
         self.queue = queue
         self.jsonEncoder = JSONEncoder()
-        jsonEncoder.dateEncodingStrategy = .iso8601
+
+        let iso8601Formatter: DateFormatter = {
+            // TODO: RUMM-437 Find a way to test encoded values
+            // Taken from https://useyourloaf.com/blog/swift-codable-with-custom-dates/
+            let formatter = DateFormatter()
+            formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS"
+            formatter.calendar = Calendar(identifier: .iso8601)
+            formatter.timeZone = TimeZone(secondsFromGMT: 0)
+            formatter.locale = Locale(identifier: "en_US_POSIX")
+            return formatter
+        }()
+        jsonEncoder.dateEncodingStrategy = .formatted(iso8601Formatter)
```

Comment:
Same should be done in `LogConsoleOutput` to also include factorial seconds in logs printed to the console if users configure the JSON format. I think we could decouple the `iso8601Formatter` to a global constant.

Reply:
done ✅